### PR TITLE
cylc.rundb: add broadcast_states table

### LINKED ISSUE #1519: cylc.rundb: new table for current broadcasts settings
The `broadcasts` table added in #1493 records all broadcast events. Although it is possible to use it to determine the current broadcast settings, it is not the easiest thing to do. We should probably rename the table as `broadcast_events` and add a new table called `broadcast_states` for the current broadcast settings (c.f. `task_events` and `task_states`).

### PR DESCRIPTION
**Pull request by matthewrmshin**
Add `broadcast_states` table for current broadcasts.
Rename table `broadcasts` to `broadcast_events`.

Close #1519.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@hjoliver @arjclark please review.

File: lib/cylc/rundb.py [hunk 1]
```diff
@@ -171,14 +163,15 @@ class CylcSuiteDAO(object):
...
+    TABLE_BROADCAST_EVENTS = "broadcast_events"
+    TABLE_BROADCAST_STATES = "broadcast_states"
     TABLE_TASK_JOBS = "task_jobs"
     TABLE_TASK_JOB_LOGS = "task_job_logs"
     TABLE_TASK_EVENTS = "task_events"
     TABLE_TASK_STATES = "task_states"
 
     TABLES_ATTRS = {
-        TABLE_BROADCASTS: [
+        TABLE_BROADCAST_EVENTS: [
```
Comment (arjclark) [inline lib/cylc/rundb.py]: Given this change, we should have something in the restart code that migrates the old table to the new one so as to be able to upgrade suite versions.

**Comment by matthewrmshin** [comment]
Back compat logic added.

**Comment by hjoliver** [comment]
I'm getting consistent failures (> 50% of the time) from ```tests/broadcast/00-simple.t```` (test 4):
```
--- select-broadcast-states.out 2015-07-10 16:38:44.000000000 +1200
+++ -   2015-07-10 16:38:44.025034245 +1200
@@ -5,7 +5,5 @@
 *|m8|[environment]BCAST|M8
 *|m9|[environment]BCAST|M9
 *|root|[environment]BCAST|ROOT
-*|wibble|[environment]BCAST|WIBBLE
 2010080900|baz|[environment]BCAST|BAZ
 2010080900|m2|[environment]BCAST|M2
-2010080900|qux|[environment]BCAST|QUX

```

**Comment by matthewrmshin** [comment]
Sorry, this should now work correctly.

**Comment by hjoliver** [comment]
Test battery passes, restarts from older version, looks good.  I'm happy with this change. Over to you @arjclark

**Comment by arjclark** [comment]
looks good to me. Tests passing in my environment.

**Pull request merged**